core/src/test/scala/org/apache/spark/sql/delta/DeltaRestoreTableSuite.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import java.io.File
+import java.sql.Timestamp
+import scala.language.implicitConversions
+
+import io.delta.implicits.DeltaDataFrameWriter
+import io.delta.tables.DeltaTable.{forPath => DeltaTableForPath}
+import org.apache.commons.io.FileUtils
+import org.apache.hadoop.fs.Path
+
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.{SQLTestUtils, SharedSparkSession}
+import org.apache.spark.sql.{DataFrame, QueryTest, SaveMode}
+
+trait DeltaRestoreTableSuiteBase extends QueryTest
+  with SharedSparkSession
+  with SQLTestUtils
+  with DeltaTestUtilsForTempViews {
+
+  private lazy val versionZeroData = spark.range(2).withSequenceColumn("seq").toDF()
+  private lazy val versionOneData = spark.range(2, 4)
+    .withSequenceColumn("seq")
+    .withColumnRenamed("id", "new_id")
+  private lazy val allDatasets = Seq(versionZeroData, versionOneData)
+
+  test("basic case - Scala restore table using version - Non-partitioned") {
```

Comment:
I think you can combine the partitioned and corresponding non-partitioned tests together into by putting it in a loop
```
for(isPartitioned <- Seq(false, true) {
   val (testNameSuffix, partitionCols = if (isPartition) ("partitioned", Some("seq")) else ("non-partitioned", None)
   test("xyz " + testNameSuffix { 
     ....
    }
}
```
reduces code duplication.